### PR TITLE
Vcf filtering

### PR DESCRIPTION
**Pull request by jkbonfield**
The workflows section was getting long, so I split it into separate documents.

I then added a VCF filtering guide, with images showing true vs false variants plotted against a range of INFO and FORMAT metrics.  It's clear some of these can work well for filtering variants, although how to do so well is tricky as the values needed are often depth specific.  The main aim is to teach people how to explore filtering on their own data sets (assuming they have a known truth set).  Possibly we should improve on this by providing some scripts, rather than the one-liners embedded within the tutorial, although that is at least educational.

For now this is a draft PR as the parameters filtered on are tied to the changes in https://github.com/samtools/bcftools/pull/1363, especially for indel filtering.

I think only once that bcftools goes live (not this release, but hopefully the one after) can we consider publishing a filtering guide.

A preformatted version of this PR (at the time of writing at least) can be seen here:

https://jkbonfield.github.io/www.htslib.org/workflow-filter/

### CODE REVIEW

**Review by pd3 — APPROVED** [review]

File: workflow-filter.md [hunk 1]
```diff
@@ -0,0 +1,493 @@
...
+-h  INT     | A coefficient for the likelihood of variations in
+homopolymer being genuine or sequencing artifact.  Lower indicates
+more likely to be an error.  The default is now 500, but used to be 100.
+--indel-bias FLOAT | A catch-all parameter to call more indels (higher
+FLOAT) at the expense of precision, or fewer (lower FLOAT) but more
+precise calls.  The default is 1.0.
+
+Additionally `bcftools call` has some options which govern output of variants.
+
+Option &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;      | Description
```
Comment (daviesrob) [inline workflow-filter.md]: Are all these non-breaking spaces necessary?

File: workflow-filter.md [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+plentiful.  We do not cover them all, but include DP, MQBZ, RPBZ,
+and SCBZ below.  Additional filtering INFO and FORMAT fields can be
+requested using the `mpileup -a` option and we also include
+FORMAT/SP.  See the man page for full details of other filtering
+fields.
+
+The most obvious filter parameter however is the QUAL field.
+
+Bcftools can filter-in or filter-out using options `-i` and `-e`
+respectively on the `bcftools view` or `bcftools filter1 commands.  For
```
Comment (daviesrob) [inline workflow-filter.md]: Missing closing back-tick here.

**Review by daviesrob — COMMENTED** [review]
`images/150x_DPQ30_lhist.png` appears to be empty (and possibly not referenced)?  Is it supposed to be there?

Rather than have `workflow-cram.md` etc, would it be possible to put the workflows into their own directory (so `workflow/cram.md)`, so the root directory doesn't get too crowded?

File: workflow-filter.md (hunk 1 above)
Comment (jkbonfield) [reply workflow-filter.md]: Yes or it looks totally rubbish, but I've found a better solution in ``.

Comment (jkbonfield) [reply workflow-filter.md]: Also removed the DPQ30 images, added the ones I missed (BQBZ), fixed the rogue quote and done some shuffling to a subdirectory.  Testing it atm.  Thanks for feedback

**Comment by jkbonfield** [comment]
After a ridiculous amount of trial and error to prevent 0 headings or 2 headings per page, I finally wrestled it into submission and I think this is now ready for review again.

**Pull request merged**